GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -103,36 +103,53 @@ - (GIDConfiguration *)configuration {
 
 - (GIDToken *)accessToken {
   @synchronized(self) {
-    if (!_cachedAccessToken) {
-      _cachedAccessToken = [[GIDToken alloc] initWithTokenString:_authState.lastTokenResponse.accessToken
-                                                  expirationDate:_authState.lastTokenResponse.
-                                                                     accessTokenExpirationDate];
-    }
+    return _cachedAccessToken;
   }
-  return _cachedAccessToken;
 }
 
 - (GIDToken *)refreshToken {
   @synchronized(self) {
-    if (!_cachedRefreshToken) {
-      _cachedRefreshToken = [[GIDToken alloc] initWithTokenString:_authState.refreshToken
-                                                   expirationDate:nil];
-    }
+    return _cachedRefreshToken;
   }
-  return _cachedRefreshToken;
 }
 
 - (nullable GIDToken *)idToken {
   @synchronized(self) {
-    NSString *idTokenString = _authState.lastTokenResponse.idToken;
-    if (!_cachedIdToken && idTokenString) {
+    return _cachedIdToken;
+  }
+}
+
+- (id<GTMFetcherAuthorizationProtocol>) fetcherAuthorizer {
+  return [_authentication fetcherAuthorizer];
```

Comment:
Let's look at creating a single `GTMAppAuthFetcherAuthorization` instance at `GIDGoogleUser` initialization time and letting it own the `OIDAuthState` data.  That instance could be stored in the property ivar and this getter would go away.

Reply:
As I mentioned in previous comment. GIDAuthentication is good encapsulation for authorization related methods.